docusaurus/docs/develop/developer_guide/adding_params.md

```diff
@@ -3,275 +3,302 @@ sidebar_position: 5
-           require.Error(t, err)
-           require.Contains(t, err.Error(), tt.expectedErr.Error())
-         } else {
-           require.NoError(t, err)
-         }
-       })
-     }
-   }
-   ```
-
-   ```go
-   func TestMsgUpdateParam_UpdateNewParameterNameOnly(t *testing.T) {
-     var expectedNewParameterName uint64 = 100
-
-     // Set the parameters to their default values
-     k, msgSrv, ctx := setupMsgServer(t)
-     defaultParams := prooftypes.DefaultParams()
-     require.NoError(t, k.SetParams(ctx, defaultParams))
-
-     // Ensure the default values are different from the new values we want to set
-     require.NotEqual(t, expectedNewParameterName, defaultParams.NewParameterName)
-
-     // Update the new parameter
-     updateParamMsg := &prooftypes.MsgUpdateParam{
-       Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
-       Name: prooftypes.ParamNewParameterName,
-       AsType: &prooftypes.MsgUpdateParam_AsInt64{AsInt64: int64(expectedNewParameterName)},
-     }
-     res, err := msgSrv.UpdateParam(ctx, updateParamMsg)
-     require.NoError(t, err)
-
-     require.Equal(t, expectedNewParameterName, res.Params.NewParameterName)
-   }
-   ```
-
-
-1. **Add Parameter Validation**
-
-   Implement a validation function for the new parameter in the same Go file.
-
-   ```go
-   func ValidateNewParameterName(v interface{}) error {
-     _, ok := v.(uint64)
-     if !ok {
-       return fmt.Errorf("invalid parameter type: %T", v)
-     }
-     return nil
-   }
-   ```
-
-1. **Add the Parameter to `ParamSetPairs()`**
-
-   Include the new parameter in the `ParamSetPairs` function.
-
-   ```go
-   func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-     return paramtypes.ParamSetPairs{
-       // Other existing parameters...
-
-       paramtypes.NewParamSetPair(
-         KeyNewParameterName,
-         &p.NewParameterName,
-         ValidateNewParameterName,
-       ),
-     }
-   }
-   ```
\ No newline at end of file
+    })
+  }
+}
+```
+
+#### 10.2 Parameter Update Tests
+
+```go
+func TestMsgUpdateParam_UpdateNewParameterNameOnly(t *testing.T) {
+  var expectedNewParameterName uint64 = 100
+
+  // Set the parameters to their default values
+  k, msgSrv, ctx := setupMsgServer(t)
+  defaultParams := prooftypes.DefaultParams()
+  require.NoError(t, k.SetParams(ctx, defaultParams))
+
+  // Ensure the default values are different from the new values we want to set
+  require.NotEqual(t, expectedNewParameterName, defaultParams.NewParameterName)
+
+  // Update the new parameter
+  updateParamMsg := &prooftypes.MsgUpdateParam{
+    Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+    Name: prooftypes.ParamNewParameterName,
+    AsType: &prooftypes.MsgUpdateParam_AsInt64{AsInt64: int64(expectedNewParameterName)},
+  }
+  res, err := msgSrv.UpdateParam(ctx, updateParamMsg)
+  require.NoError(t, err)
+
+  require.Equal(t, expectedNewParameterName, res.Params.NewParameterName)
```

Comment:
Wdyt about adding a comment that says that this test should also assert that all other params of the same module remain unchanged. See: https://github.com/pokt-network/poktroll/blob/main/x/proof/keeper/msg_server_update_param_test.go#L39.